### DISCUSSION TITLE
What is the proper way to read a number (int) from a binary file?

### DISCUSSION THREAD

**Discussion by elgs** [denoland/deno#9652] [Q&A] [answered]
I want to:
1. seek to position p from the beginning of a binary file;
2. read the next 4 bytes;
3. and turn what's read into a number (32bit integer);

I can do the first with Deno.seek, but Deno.read doesn't provide a parameter for how many bytes to read. So how can I do it? Thanks.

**Comment by caspervonb** [comment, marked as answer]
> but Deno.read doesn't provide a parameter for how many bytes to read.

It does, it can only read up to N bytes where the number of bytes is the byteLength of the TypedArray passed in.

- Seek to the offset,
- Pass an Uint8Array with a byteLength of 4 to Deno.read
- Use a DataView and call getInt32(0) to read that back as i32 (getUint32 for u32).

If you have any more questions you can find us in the #help channel on Discord https://discord.gg/bpHe9MVB

Or if you prefer the forum format, please use the [Discussions](https://github.com/denoland/deno/discussions/categories/q-a) tab next time for general Q&A as the issue tracker is meant for bugs 👍

**Comment by elgs** [comment]
Thank you so much!

**Comment by kbridge** [comment]
You can use `readRange` from `std/io/files.ts`. ([Documentation](https://deno.land/std@0.148.0/io/files.ts?s=readRange)) It encapsulates the seeking to the offset and reading N bytes to a `Uint8Array` steps. ([Source Code](https://deno.land/x/std@0.148.0/io/files.ts?code#L31)) It's the same as the first two steps of @caspervonb's answer, so it's equivalent but saves some boilerplate code.
